cxx/clean_relation.hh

```diff
@@ -155,17 +155,28 @@ class CleanRelation : public Relation<T> {
     return z;
   }
 
-  double cluster_or_prior_logp(std::mt19937* prng, const T_items& z,
+  double cluster_or_prior_logp(std::mt19937* prng, const T_items& items,
                                const ValueType& value) const {
-    if (clusters.contains(z)) {
-      return clusters.at(z)->logp(value);
+    if (clusters.contains(items)) {
+      return clusters.at(items)->logp(value);
     }
     Distribution<ValueType>* prior = make_new_distribution(prng);
     double prior_logp = prior->logp(value);
     delete prior;
     return prior_logp;
   }
 
+  ValueType cluster_or_prior_sample(std::mt19937* prng,
```

Comment:
I renamed it to `sample_at_items`, since "for" sounded a little ambiguous to me. I liked how `cluster_or_prior_sample` was the corresponding sampling method for `cluster_or_prior_logp` but I agree it's clunky.

Agreed re. RNG, I added a TODO.